classes/helper.php

```diff
@@ -371,4 +374,78 @@ public static function regular_expression_search(string $search, string $table,
         }
         return $results;
     }
+
+    /**
+     * Get column info from column name.
+     *
+     * @param string $table The table name.
+     * @param string $columnname The column name.
+     */
+    public static function get_column_info(string $table, string $columnname): database_column_info {
+        global $DB;
+
+        $columns = $DB->get_columns($table);
+        $column = null;
+        foreach ($columns as $col) {
+            if ($col->name == $columnname) {
+                $column = $col;
+                break;
+            }
+        }
+
+        if (is_null($column)) {
+            throw new moodle_exception(get_string('errorcolumnnotfound', 'tool_advancedreplace', $columnname));
+        }
+
+        return $column;
+    }
+
+    /**
+     * A clone of the core function replace_all_text.
+     * We have optional id parameter to restrict the search.
+     *
+     * @since Moodle 2.6.1
+     * @param string $table name of the table
+     * @param database_column_info $column
+     * @param string $search
+     * @param string $replace
+     * @param int $id optional id to restrict the search
+     */
+    public static function replace_all_text($table, database_column_info $column, $search, $replace, $id = null) {
```

Comment:
when we land this in core I think it would be more useful to expand to to a generic where clause not just locked to id, so lets do this now we its in the right shape

Reply:
wheresql and params are added